### PR TITLE
fix(DB/creature): Improve - The Battle For The Ebon Hold

### LINKED ISSUE #644: DB/Quest: The Battle For The Ebon Hold
**Description**: 
-the knights of death with us do not attack enimies when they approach them.
-the abominations instantly throw their hooks

**Current behaviour**: Tell us what happens
-the knights of death with us do not attack enimies when they approach them.
-the abominations instantly throw their hooks

**Expected behaviour**: Tell us what should happen instead
-the DK must attack when enemies approching
-the abominations must cast their hook like video below
https://www.youtube.com/watch?v=QHWpZexZODY

**Steps to reproduce the problem**:
```
.go creature id 31084
.gm off
.mod phase 488 (use this even if you do .gm on and .gm off in a second time)
.quest add 13166
```
- Use the purple platform to go downstairs and do the quest

**Branch(es):** Master

**AC hash/commit**: 9d2395e

**Operating system:** Ubuntu server 16.04.03 LTS

**Modules:** Are you using modules? if yes, list them

**Other Customizations:** Are you using any core/db patches or other customizations? If yes please specify them here.

[//]: # (This template is for problem reports, for other type of reports edit it accordingly)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)
[//]: # (For fixes containing c++ create a Pull Request)

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/49387499-db-quest-the-battle-for-the-ebon-hold?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Si1ker**
Improve creatures involved in the quest "The Battle For The Ebon Hold"

References:
https://www.youtube.com/watch?v=FkKoxtykUn0
https://www.youtube.com/watch?v=QHWpZexZODY
https://www.youtube.com/watch?v=gf59hSeSolU  (enrage 6:45)
https://www.youtube.com/watch?v=Y37Dm5xQeSQ

## CHANGES PROPOSED:
-  Update level range for Val'kyr Battle-maiden (59-60)
-  Update Terrifying Abomination wander distance
- Update level range for Terrifying Abomination (57-58)
- Fix Terrifying Abomination's SAI
- Update SAI, unit_flag and update level range for Knight of the Ebon Blade (57-58)
=> I am setting an absolute unit_flag and I am not going to add as the current one in DB is totally wrong.
- Add missing Knight of the Ebon Blade corpses on lower level
- Update Knight of the Ebon Blade positions
- Add Patchwerk texts
- Add translations for Spanish
- Update AIName to Patchwerk and build SAI
- Make spell 'Scourge hook' to only target players

## ISSUES ADDRESSED:
- Closes #644 

## TESTS PERFORMED:
- Tested in-game
- /* Affected rows: 23  Found rows: 0  Warnings: 0  Duration for 17 queries: 0.016 sec. */

## HOW TO TEST THE CHANGES:

== If you do not have a DK on that part of the quest chain =
.gm on
.go c 130488
check

== If you have a DK on that part of the quest chain =
Get the quest from Darion -> [The Battle For The Ebon Hold](https://www.wowhead.com/quest=13166/the-battle-for-the-ebon-hold)
Enjoy !

## KNOWN ISSUES AND TODO LIST:

- Patchwerk emotes/texts need improvement

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Helias** [comment]
tested, it works

image

File: data/sql/updates/pending_db_world/rev_1606590908424148300.sql [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+UPDATE `creature_template` SET `minlevel`=57 WHERE `entry`=31098;
+
+-- Fix Terrifying Abomination's SAI
+DELETE FROM `smart_scripts` WHERE `entryorguid`=31098 AND `source_type`=0;
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
+(31098, 0, 0, 0, 9, 0, 50, 1, 8, 20, 0, 0, 0, 11, 50335, 1, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Terrifying Abomination - Within 8-20 Range - Cast Scourge Hook'),
+(31098, 0, 1, 0, 0, 0, 80, 0, 0, 0, 7000, 9000, 0, 11, 15496, 1, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Terrifying Abomination - In Combat - Cast Cleave');
+
+-- Update SAI, unit_flag and update level range for Knight of the Ebon Blade (57-58)
+UPDATE `creature_template` SET `minlevel`=57, `unit_flags`=557056, `AIName`='SmartAI' WHERE `entry`=31094;
```
Comment (Helias) [inline data/sql/updates/pending_db_world/rev_1606590908424148300.sql:31]: change the flags in this way
https://www.azerothcore.org/wiki/Dealing-with-SQL-files

**Review by Helias — DISMISSED** [review]

File: data/sql/updates/pending_db_world/rev_1606590908424148300.sql (hunk 1 above)
Comment (Si1ker) [reply data/sql/updates/pending_db_world/rev_1606590908424148300.sql:31]: check changes proposed section

**Comment by Voxstrasza** [comment]
image

**H Y P E**

looks really good! 👍🏻

File: data/sql/updates/pending_db_world/rev_1606590908424148300.sql [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+/*
+ * Update by Silker | github.com/Si1ker
+ * Copyright (C) | www.azerothcore.org
+*/
+
+/*
+References:
+https://www.youtube.com/watch?v=FkKoxtykUn0
+https://www.youtube.com/watch?v=QHWpZexZODY
+*/
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1606590908424148300.sql:12]: no need to put this stuff here, it's better in the PR description

**Comment by iyasenai** [comment]
It works, watch the video :)
https://www.youtube.com/watch?v=sVR3VkFqXIE

**Comment by Si1ker** [comment]
Thanks @iyasenai , nice test as always ! 🚀

**Review by FrancescoBorzi — APPROVED** [review]

File: data/sql/updates/pending_db_world/rev_1606590908424148300.sql [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+UPDATE `creature` SET `position_x`=2410.179, `position_y`=-5621.223, `position_z`=420.653 WHERE `guid`=130488;
+UPDATE `creature` SET `position_x`=2411.769, `position_y`=-5624.546, `position_z`=420.652 WHERE `guid`=130485;
+UPDATE `creature` SET `position_x`=2413.665, `position_y`=-5627.342, `position_z`=420.652 WHERE `guid`=130487;
+UPDATE `creature` SET `position_x`=2415.735, `position_y`=-5629.859, `position_z`=420.651 WHERE `guid`=130486;
+
+-- Add Patchwerk texts
+DELETE FROM `creature_text` WHERE `CreatureID`=31099 AND `GroupID`=0;
+INSERT INTO `creature_text` (`CreatureID`, `GroupID`, `ID`, `Text`, `Type`, `Language`, `Probability`, `Emote`, `Duration`, `Sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES 
+(31099, 0, 0, '%s goes into a frenzy!', 41, 0, 100, 0, 0, 0, 38630, 0, 'Patchwerk emote Frenzy - Ebon Hold');
+/* Add translations */
```
Comment (Winfidonarleyan) [inline data/sql/updates/pending_db_world/rev_1606590908424148300.sql:56]: ???
Use only `broadcast_text_locale`
image

**Pull request merged**